File: java/OfferWallExample/app/src/main/java/com/pubscale/sdkone/example/offerwallexample/MainActivity.java
package com.pubscale.sdkone.example.offerwallexample;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.pubscale.sdkone.example.offerwallexample.databinding.ActivityMainBinding;
import com.pubscale.sdkone.offerwall.OfferWall;
import com.pubscale.sdkone.offerwall.OfferWallConfig;
import com.pubscale.sdkone.offerwall.models.OfferWallInitListener;
import com.pubscale.sdkone.offerwall.models.OfferWallListener;
import com.pubscale.sdkone.offerwall.models.Reward;
import com.pubscale.sdkone.offerwall.models.errors.InitError;

public class MainActivity extends AppCompatActivity {

    private SharedPreferences sharedPreferences;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        sharedPreferences = getSharedPreferences("ow_example", Context.MODE_PRIVATE);

        String balance = String.valueOf(sharedPreferences.getFloat("balance", 0f));
        binding.tvBalance.setText(balance);

        binding.init.setOnClickListener(view -> {
            setupGGOfferWall();
        });

        binding.launch.setOnClickListener(view -> {
            OfferWall.launch(this, ggOfferWallListener);
        });

        binding.destroy.setOnClickListener(view -> {
            OfferWall.destroy();
            binding.destroy.setEnabled(false);
        });
    }

    private void setupGGOfferWall() {
        String uniqueIdTyped = binding.uniqueIdText.getText().toString();
        String appKey = "f53e007b-a86c-436d-a7d3-40176ccab173";

        OfferWallConfig offerWallConfig = new OfferWallConfig.Builder(getApplicationContext(), appKey)
                .setTitle("Title of your offer wall")
                .setUniqueId(uniqueIdTyped)
                .build();

        OfferWall.init(offerWallConfig, new OfferWallInitListener() {
            @Override
            public void onInitSuccess() {
                binding.destroy.setEnabled(true);
                Toast.makeText(getApplicationContext(), "OfferWall SDK Initialed", Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onInitFailed(InitError initError) {
                Toast.makeText(getApplicationContext(), "OfferWall SDK Init Failed: "+ initError, Toast.LENGTH_SHORT).show();
            }
        });
    }

    private OfferWallListener ggOfferWallListener = new OfferWallListener() {
        @Override
        public void onOfferWallShowed() {
            Toast.makeText(getApplicationContext(), "Offer wall showed", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onOfferWallClosed() {
            Toast.makeText(getApplicationContext(), "Offer wall closed", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onRewardClaimed(Reward reward) {
            float balance = sharedPreferences.getFloat("balance", 0f);
            balance += reward.getAmount();

            binding.tvBalance.setText(String.valueOf(balance));
            sharedPreferences.edit().putFloat("balance", balance).apply();

            Toast.makeText(
                    getApplicationContext(),
                    "Offer wall reward claimed: " + reward.getAmount() + " " + reward.getCurrency(),
                    Toast.LENGTH_SHORT
            ).show();
        }

        @Override
        public void onFailed(String message) {
            Toast.makeText(getApplicationContext(), "onFailed: " + message, Toast.LENGTH_SHORT).show();
        }
    };
}

File: kotlin/OfferWallExample/app/src/main/java/com/pubscale/sdkone/example/offerwallexample/MainActivity.kt
package com.pubscale.sdkone.example.offerwallexample

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.pubscale.sdkone.example.offerwallexample.databinding.ActivityMainBinding
import com.pubscale.sdkone.offerwall.OfferWall
import com.pubscale.sdkone.offerwall.OfferWallConfig
import com.pubscale.sdkone.offerwall.models.OfferWallInitListener
import com.pubscale.sdkone.offerwall.models.OfferWallListener
import com.pubscale.sdkone.offerwall.models.Reward
import com.pubscale.sdkone.offerwall.models.errors.InitError

class MainActivity : AppCompatActivity() {

    private lateinit var sharedPreferences: SharedPreferences
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        sharedPreferences = getSharedPreferences("ow_example", Context.MODE_PRIVATE)

        val balance = sharedPreferences.getFloat("balance", 0f).toString()
        binding.tvBalance.text = balance

        binding.init.setOnClickListener {
            setupGGOfferWall()
        }

        binding.launch.setOnClickListener {
            OfferWall.launch(this, ggOfferWallListener)
        }

        binding.destroy.setOnClickListener {
            OfferWall.destroy()
            binding.destroy.isEnabled = false
        }
    }

    private fun setupGGOfferWall() {
        val uniqueIdTyped = binding.uniqueIdText.text.toString()
        val appKey = "f53e007b-a86c-436d-a7d3-40176ccab173"

        val offerWallConfig = OfferWallConfig.Builder(this, appKey)
            .setTitle("Title of your offer wall")
            .setUniqueId(uniqueIdTyped)
            .build()

        OfferWall.init(offerWallConfig, object: OfferWallInitListener {
            override fun onInitSuccess() {
                binding.destroy.isEnabled = true
                Toast.makeText(this@MainActivity, "OfferWall SDK Initialed", Toast.LENGTH_SHORT).show()
            }

            override fun onInitFailed(error: InitError) {
                Toast.makeText(this@MainActivity,"OfferWall SDK Init Failed: $error",Toast.LENGTH_SHORT).show()
            }
        })
    }

    private val ggOfferWallListener = object: OfferWallListener {
        override fun onOfferWallShowed() {
            Toast.makeText(this@MainActivity, "Offer wall showed", Toast.LENGTH_SHORT).show()
        }

        override fun onOfferWallClosed() {
            Toast.makeText(this@MainActivity, "Offer wall closed", Toast.LENGTH_SHORT).show()
        }

        override fun onRewardClaimed(reward: Reward) {
            var balance = sharedPreferences.getFloat("balance", 0f)
            balance += reward.amount

            binding.tvBalance.text = balance.toString()
            sharedPreferences.edit().putFloat("balance", balance).apply()

            Toast.makeText(
                this@MainActivity,
                "Offer wall reward claimed: ${reward.amount} ${reward.currency}",
                Toast.LENGTH_SHORT
            ).show()
        }

        override fun onFailed(message: String) {
            Toast.makeText(this@MainActivity, "onFailed: $message", Toast.LENGTH_SHORT).show()
        }
    }
}